File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.c
int lengthOfLongestSubstring(char* s) {
        int *freq=(int*)malloc(256*sizeof(int));
        for(int i=0;i<256;i++)
        {
            freq[i]=0;
        }
        int j=0;
        int i=0;
        int max_win=0;
        while(j<strlen(s))
        {
            while(freq[s[j]]==1)
            {
                freq[s[i]]=0;
                i++;
            }
            freq[s[j]]=1;
            int cur_win=j-i+1;
            max_win=MAX(max_win,cur_win);
            j++;
        }
        return max_win;
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] freq=new int[256];
        for(int i=0;i<256;i++)
        {
            freq[i]=0;
        }
        int j=0;
        int i=0;
        int max_win=0;
        while(j<s.length())
        {
            while(freq[s.charAt(j)]==1)
            {
                freq[s.charAt(i)]=0;
                i++;
            }
            freq[s.charAt(j)]=1;
            int cur_win=j-i+1;
            max_win=Math.max(max_win,cur_win);
            j++;
        }
        return max_win;
    }
}